### PR TITLE
Implement ItemBlockAction plugin for ServiceAccounts

### PR DESCRIPTION
**Pull request by sseago**
Returns related SCCs for the ServiceAccount

### CODE REVIEW

File: velero-plugins/serviceaccount/itemblock.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to everything.
+func (p *IBAPlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"serviceaccounts"},
+	}, nil
+}
+
+// Execute copies local registry images into migration registry
```
Comment (kaovilai) [inline velero-plugins/serviceaccount/itemblock.go]: Comment needs updating.

**Review by kaovilai — CHANGES_REQUESTED** [review]

**Comment by sseago** [comment]
@kaovilai Updated the comment

File: velero-plugins/serviceaccount/backup.go [hunk 2]
```diff
@@ -100,8 +107,8 @@ func (p *BackupPlugin) UpdateSCCMap() error {
 				if namespace == "" {
 					continue
 				}
-				if p.SCCMap[namespace] == nil {
-					p.SCCMap[namespace] = make(map[string][]apisecurity.SecurityContextConstraints)
+				if c.SCCMap[namespace] == nil {
+					c.SCCMap[namespace] = make(map[string][]apisecurity.SecurityContextConstraints)
```
Comment (kaovilai) [inline velero-plugins/serviceaccount/backup.go:111]: Do we need eventual deletion of items from the map somewhere? I expect this map will grow indefinitely ATM?

Comment (sseago) [reply velero-plugins/serviceaccount/backup.go:111]: @kaovilai Actually each time we repopulate this we get the full list of SCCs in-cluster, and then populate all SAs for all namespaces which map to it. So the size should only grow as the number of SCCs and related SAs grows. We update it once for a backup to make sure it's up-to-date as of backup start time. Realistically, we probably don't need to track by backup since the plugin currently exits after backup processing, so next backup starts with an empty map anyway, but verifying per backup makes the plugin resiliant to that design choice.

tl;dr this map shouldn't grow -- it should always be the size of the actual in-cluster mapping between all SAs and all SCCs.

Comment (kaovilai) [reply velero-plugins/serviceaccount/backup.go:111]: I guess I need to look at how plugin exits.. and if it kills/clears all memory usage.

Comment (sseago) [reply velero-plugins/serviceaccount/backup.go:111]: @kaovilai There's a cleanup call made at the end of the the reconcile loop that kills the plugin process. We had a Velero bug a few months ago where in some cases this wasn't happening properly, and plugin processes piled up (that's fixed now, though). Easy enough to confirm by watching processes in the pod while running a backup. If the process exits, surely that means it's no longer consuming memory.

**Review by kaovilai — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]
